src/firmware/app/control/trajectory_planner.h

```diff
@@ -56,6 +35,44 @@ typedef struct FirmwareRobotPathParameters
     float final_speed;
 } FirmwareRobotPathParameters_t;
 
+// Struct that defines a single point on a trajectory
+// Includes the Position and Time data corresponding to that point
+typedef struct PositionTrajectoryElement
+{
+    Vector2d_t position;
+    float time;
+} PositionTrajectoryElement_t;
+
+typedef struct PositionTrajectory
+{
+    PositionTrajectoryElement_t* trajectory_elements;
+    FirmwareRobotPathParameters_t path_parameters;
+    float* speed_profile;
+} PositionTrajectory_t;
+
+// Struct that defines a single point on a velocity trajectory
+// Includes the velocity and Time data corresponding to that point
+typedef struct VelocityTrajectoryElement
+{
+    Vector2d_t linear_velocity;
+    float angular_velocity;
+    float time;
+} VelocityTrajectoryElement_t;
+
+typedef struct VelocityTrajectory
+{
+    VelocityTrajectoryElement_t* trajectory_elements;
+    FirmwareRobotPathParameters_t path_parameters;
+} VelocityTrajectory_t;
+
+typedef enum TrajectoryPlannerGenerationStatus
+{
+    OK                              = 0,
+    FINAL_VELOCITY_TOO_HIGH         = 1,
+    INITIAL_VELOCITY_TOO_HIGH       = 2,
+    INTERPOLATION_ELEMENT_MAXED_OUT = 3,
```

Comment:
Might need a comment for this status, IMO it's not as obvious as the others

Reply:
Added explanatory comment 👍